engine/src/main/java/org/terasology/config/flexible/FlexibleConfig.java

```diff
@@ -54,4 +59,33 @@
      * Returns a boolean stating whether the config contains a {@link Setting} with the given id.
      */
     boolean contains(SimpleUri id);
+
+    /**
+     * Returns a map of all the settings, allowing iteration of all the settings.
+     *
+     * @return A map containing all the settings, along with their id.
+     */
+    Map<SimpleUri, Setting> getSettings();
+
+    /**
+     * Saves the values of all settings having non-default values, to enable persistence across sessions.
+     *
+     * All the non-default values that were not used in this session and are still "parked" are also
+     * saved as-is, to be used later.
+     *
+     * @param writer A writer that will serve as the destination of settings.
+     */
+    void save(Writer writer);
+
+    /**
+     * Loads the values of the settings having non-default values, to enable persistence across sessions.
+     *
+     * All the non-default values are loaded and "parked" till the actual setting is added to the config,
+     * and which point they are "unparked" and parsed, replacing the default value of the setting.
+     *
+     * Note that this function should be called before adding any settings to the FlexibleConfig.
+     *
+     * @param reader A reader that will serve as the source of the settings.
+     */
+    void load(Reader reader);
```

Comment:
I would separate loading and saving from managing settings. Changing and reading setting values and load/save are two different responsibilities for different clients and the strategy how to save a setting (io.Reader and io.Writer) are implementation details which should not be part of this interface.

Reply:
Actually, I had the loading/saving login in FCManager before, but @emanuele3d commented the following:
> I am thinking that in terms of load/save functionality, it makes sense that the manager orchestrates it, i.e. iterating over the FCs as it happens here and providing the file or even the reader. But I'd want the FCs themselves to handle the actual reading/writing, to separate the concerns and mantain some of the functionality private to the FC class.
Which is why I put this here.

I'm fine with either, so can you two decide which feels better? :D

Reply:
I'd keep it as it is for the time being. Let's get this PR moving again and merged in. We can always refactor after we have gained some experience with FCs in general.